utils/lib/DataSource.ts

```diff
@@ -70,6 +100,10 @@ class DataSource extends Component<DataSourceConfig, string> {
   }
 
   getConfigContent() {
+    if (this.isCoreDataSource) {
+      return { id: this.identifier } as DataSourceConfig;
+    }
```

Comment:
Same comment here as the `getConfigFilePath` method. 
It seems like we're doing gymnastics just to let the DataSource class support being a "core" data source. Originally, these classes we're created to represent the physical files in the repository, so I'm wondering if it would be better to handle the use of core data source entirely at the quickstart level and not even create a DataSource instance for them.

Reply:
I agree with this, since `core` data sources don't even really live in this repository.

I think it would be better to handle core data sources entirely at the quickstart level, but that would add some weirdness into the generalized nature of the class (`getComponents` for example)

I think it may be out of scope for this ticket, but happy to discuss further or do a follow-up ticket to help move along IP -> DS

Reply:
To add on from something @MichelLosier and i discussed when we originally were brainstorming on this (sorry for paraphrasing, its been a little bit since the initial convo) -

We kept coming back to the idea that a Data Source should be seperated from the `Component` structure and have it live individually _along_ the quickstart. Since a quickstart config only cares about the ID of a data source for mutation purposes.

Reply:
A possible idea:
Does `getComponents` even need to return the core data sources? Could they live separately from the other "components" in the Quickstart class?